lib/type1.g

```diff
@@ -83,37 +83,61 @@ InstallAttributeFunction(
 ##  <#/GAPDoc>
 ##
 Subtype := "defined below";
-
+if IsHPCGAP then
+    DS_TYPE_CACHE := ShareSpecialObj([]);
+fi;
 
 BIND_GLOBAL( "NEW_FAMILY",
     function ( typeOfFamilies, name, req_filter, imp_filter )
-    local   type, pair, family;
+    local   lock, type, pair, family;
 
     # Look whether the category of the desired family can be improved
     # using the categories defined by 'CategoryFamily'.
     imp_filter := WITH_IMPS_FLAGS( AND_FLAGS( imp_filter, req_filter ) );
     type := Subtype( typeOfFamilies, IsAttributeStoringRep );
+    if IsHPCGAP then
+        lock := READ_LOCK(CATEGORIES_FAMILY);
+    fi;
     for pair in CATEGORIES_FAMILY do
```

Comment:
Perhaps mention in a comment here that you are manually calling READ_LOCK because the compiler doesn't support atomic? (Assuming my guess is right. If not, add comment for why we aren't using atomic here :) )

Reply:
That's precisely why, *but* this has been in those `*1.g` for a long time, i.e. it is not something added by this PR. While I think documenting this is the right thing, I really don't want to redo this PR once again (and wait for tests etc.) just for adding a comment that somebody else should have added a couple years ago...